src/window.c

```diff
@@ -156,6 +156,22 @@ log_frame_layout(frame_T *frame)
 }
 #endif
 
+/*
+ * Check if the current window is allowed to move to a different buffer.
+ * If the window has 'stickybuf', then forceit must be TRUE or this function
+ * will return FALSE.
+ */
+int is_allowed_to_go_to_buffer(int forceit)
```

Comment:
```suggestion
int check_can_go_to_buffer(int forceit)
```

Reply:
Since `goto` is used a lot in this repository and this function doesn't have much to do with that, I'd like to avoid potentially confusing people. How about `check_can_change_buffer`?

Reply:
Ah, it doesn't accept a buffer argument `check_can_switch_buffer` then.

Reply:
Do you think `check_can_switch_buffer` is too close to `'switchbuf'`? I'm happy to go with `check_can_switch_buffer` if that works for you

Reply:
Maybe `check_can_set_curbuf`?

Reply:
I dig it, let's go with `check_can_set_curbuf`